TwinePlatform/twine-monolith
# 🔄 switch tslint for eslint

**DO NOT MERGE**
**MERGE #307 FIRST**
fixes #32 

### Changes
- Remove `tslint-config-twine`
- Create `eslint-config-twine`
- Use `eslint-config-twine` in API
- Upgrade `react-scripts` to v3 in dashboard-app so it can use built-in linter
- Make linting updates
  - Mostly consist of fixing `react-hooks/exhaustive-deps` issues. 
  - Fixing one of these causes a render loop, fixed by adding a conditional update
- Disable `react-hooks/exhaustive-deps` for hooks where we're passing in dependency arrays from the outside.

### Testing Requirements
Just to establish nothing's broken.
  - Observe dashboard renders correctly
  - On each of `/time`, `/activity` and `/volunteer`
  - Fiddle around with the legend, unit toggle and date range
  - Go to table view, sort on various columns, toggle the sorting

### Release Requirements
- N/A

### Manual Deployment
- N/A
